src/flamegraph.cpp

```diff
@@ -195,7 +208,9 @@ QString FrameGraphicsItem::description() const
         }
         totalCost = item->cost();
     }
-    const auto symbol = m_symbol.symbol.isEmpty() ? QObject::tr("??") : m_symbol.prettySymbol;
+    const auto symbol = m_symbol.symbol.isEmpty()
```

Comment:
wrap this code in a helper function that, given a Data::Symbol, returns either the pretty symbol or not, based on a global settings singleton that's queried

Reply:
or ?? if the symbol is invalid

Reply:
I did an attempt at this. Please have a look and let me know what you think. Note that BottomUp, TopDown and CallerCallee results pages are not yet updated when the UI setting is toggled. Some pointers on how to correctly invalidate the models there would be helpful.